snapcraft/plugins/_plugin_finder.py

```diff
@@ -88,6 +88,8 @@ def get_plugin_for_base(plugin_name: str, *, build_base: str) -> PluginTypes:
     # from the plugins
     if build_base in ("core", "core16", "core18"):
         build_base = "legacy"
+    if build_base not in ("legacy", "core20"):
```

Comment:
I think it would be appropriate to force the user to specify `build-base` in a case like this.  I don't think Snapcraft should assume core20 if left unspecified.

The schema or sanity checks should require `build-base` if `base` is not something we have build image support for.

Reply:
`build-base` is always specified at this point.

The intention of this hunk of code is that, if base is not known use the highest known set of plugins, currently the core20 ones.

As otherwise, currently with `build-base: core22` the dictionary has no plugins for `core22`, and hence the build fails with "plugin nil not found". 

Or do you want me to start core22 dictinary key which is a copy of core20?

Reply:
Note this is not really a `build-base`, but rather keys of `_PLUGINS` dictionary. Of which there are only two at the moment `legacy` and `core20`. Which doesn't quite make sense to me.

Maybe I should rework this a bit? to rename keys in the `_PLUGINS` dictionary to be `v1` and `v2`, and then just use modules_path at this point? rather than remap, for the second time, which build-base is using what?

Reply:
See https://github.com/snapcore/snapcraft/pull/3338/files if that makes more sense..... Given that we have many-to-one mapping between the set of bases and plugin versions. It doesn't make sense to name the keys in the `_PLUGINS` dictinary after bases.

Or at least, it should become default-dict type, to return `core20` / `v2` ones for unknown bases?

Reply:
Ah yes, that's what I get for commenting before having my coffee :D I agree,  using keys "v1" and "v2" would be more ideal at this point rather than legacy/core20.

Reply:
@cjp256 please re-review the renames of things. And all the tests now fixed up and pass.